source/dubregistry/repositories/gitea.d

```diff
@@ -0,0 +1,284 @@
+/**
+	Copyright: © 2023 Sönke Ludwig
+	License: Subject to the terms of the GNU GPLv3 license, as written in the included LICENSE.txt file.
+	Authors: Sönke Ludwig
+*/
+module dubregistry.repositories.gitea;
+
+import dubregistry.cache;
+import dubregistry.dbcontroller : DbRepository;
+import dubregistry.repositories.repository;
+import std.string : startsWith;
+import std.typecons;
+import vibe.core.log;
+import vibe.core.stream;
+import vibe.data.json;
+import vibe.http.client : HTTPClientRequest;
+import vibe.inet.url;
+
+class GiteaRepositoryProvider : RepositoryProvider {
+	private {
+		string m_token;
+		string m_url;
+	}
+@safe:
+
+	private this(string token, string url)
+	{
+		m_token = token;
+		m_url = url.length ? url : "https://gitea.com/";
```

Comment:
a lot of gitea instances are self-hosted, could we somehow make it so that this URL can be set per-package?

Reply:
actually nvm, since you need authentication I don't think that's gonna solve this here. Does this just exist for custom/private DUB registry instances?

Reply:
> Does this just exist for custom/private DUB registry instances?

Exactly. In fact, I haven't really tested this with the official gitea.com, but with a local instance.